code/modules/hydroponics/plant_genes.dm

```diff
@@ -307,14 +307,21 @@
 
 /datum/plant_gene/trait/teleport/on_squash(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
 	if(isliving(target))
-		var/teleport_radius = max(round(G.seed.potency / 10), 1)
 		var/turf/T = get_turf(target)
+		var/area/A = get_area(T)
+		if(!T || !is_teleport_allowed(T.z) || A.tele_proof)
+			return
+		var/teleport_radius = max(round(G.seed.potency / 10), 1)
```

Comment:
This is really snowflakey, when these all use the `do_teleport` proc.

That should handle the logic for teleport proofing, not every object that teleports. Otherwise, you're copy-pasting code everywhere, which is a violation of our coding guidelines.